### ISSUE TITLE
mwc-button text color custom property does nothing

### ISSUE THREAD

**Issue by isochronous** [material-components/material-components-web-components#1176]
**Describe the bug**
I'm attempting to style a mwc-button so that it has a white background and dark grey text (and an outline in some cases). I have no trouble customizing the background color and outline color, but the `--mdc-theme-on-primary` custom property that's supposed to change the text color does absolutely nothing.

**To Reproduce**
1. Create a web component (we're using Lit-Element) that uses two mwc-buttons somewhere.  One should be an "outlined" button, the other should be a totally standard one with just a label.
2. In the styles for your web component, add the following rule: `mwc-button { --mdc-theme-primary: #ffffff; --mdc-button-outline-color: #444444; --mdc-theme-on-primary: #444444; }
3. Put your web component into a demo page, start the dev server, and look at the page.
4. Notice that the buttons have a white background, and the outlined button has the correct color outlines, but none of the text is visible because it ignores the custom property and just uses the default white text.

**Expected behavior**
The buttons' text color should respect the `--mdc-theme-on-primary` custom property, and be dark grey instead of white.

**Screenshots**
[Here's a screenshot that demonstrates the issue](https://imgur.com/Ss9sC6g)

**Browser Version (please complete the following information):**
 - OS: Windows 10
 - Browser: Chrome
 - Version 80.0.3987.106

**Additional context**
N/A

**Comment by isochronous** [comment]
Looks like this is related to #5349

**Comment by asyncLiz** [comment]
This is working as intended. Flat and outlined buttons use `--mdc-theme-primary` as their foreground color and outline color (the outline color can additionally be overridden if you want it to be different than the primary theme color).

Only the raised button (which uses the theme color as its background) uses the on-theme color as its foreground color.

The solution is to use different css selectors for your button variants. We would recommend using an attribute selector.

```css
mwc-button[raised] {
  --mdc-theme-primary: white;
  --mdc-theme-on-primary: #444;
}

mwc-button, /* don't forget about flat */
mwc-button[outlined] {
  --mdc-theme-primary: #444;
}
```

https://jsbin.com/xudirihuce/edit?html,output

**Comment by isochronous** [comment]
Yeah, I actually just came to the same conclusion by looking at the markup/styling for demos.  But in that case, your documentation really needs to have a little more detail added to specify those important distinctions, because I just wasted six hours trying to figure out what I was doing wrong when I was just following your docs.  [Just look at this screenshot](https://i.imgur.com/tbO2K3P.png) of how the custom properties are described in the `mwc-button` package readme.

I'll copy and paste the two I'm mainly talking about:
"**Name**: --mdc-theme-primary.  **Description**: Background color of the button."
and 
"**Name**: --mdc-theme-on-primary. **Description**: Text color of the button."
